renderer/components/Home/WalletsMenu.js

```diff
@@ -1,47 +1,63 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { intlShape, injectIntl } from 'react-intl'
-import { withRouter, Link } from 'react-router-dom'
+import { withRouter } from 'react-router-dom'
 import { Box } from 'rebass'
-import { Text, MenuItem } from 'components/UI'
+import { Text, Menu } from 'components/UI'
 import { WalletName } from 'components/Util'
 import messages from './messages'
 
-const WalletGroup = withRouter(({ location, setActiveWallet, title, wallets, ...rest }) => (
-  <Box p={2} {...rest}>
-    <Text
-      css={`
-        text-transform: uppercase;
-      `}
-      fontWeight="normal"
-      mb={2}
-      px={3}
-    >
-      {title}
-    </Text>
-    {wallets.map(wallet => (
-      <Link
-        key={wallet.id}
-        style={{
-          display: 'block',
-          whiteSpace: 'nowrap',
-          overflow: 'hidden',
-          textOverflow: 'ellipsis',
-        }}
-        to={`/home/wallet/${wallet.id}`}
-      >
-        <MenuItem
-          isActive={location.pathname.startsWith(`/home/wallet/${wallet.id}`)}
-          onClick={() => setActiveWallet(wallet.id)}
-        >
-          <WalletName wallet={wallet} />
-        </MenuItem>
-      </Link>
-    ))}
-  </Box>
-))
+const WalletGroupHeader = props => (
+  <Text
+    css={`
+      text-transform: uppercase;
+    `}
+    fontWeight="normal"
+    mb={2}
+    px={3}
+    {...props}
+  />
+)
+
+const WalletGroup = withRouter(
+  ({ location, history, activeWallet, setActiveWallet, title, wallets, ...rest }) => {
+    /**
+     * generateMenuItemId - Generate a menu item id from a wallet id.
+     *
+     * @param  {string} walletId Wallet Id
+     * @returns {string} Wallet menu item id
+     */
+    const generateMenuItemId = walletId => `wallet-menu-item-${walletId}`
+
+    /**
+     * generateMenuItems - Gernate a set of menu items from a list of wallets.
+     *
+     * @param  {Array} wallets List of wallets
+     * @returns {Array} List of menu items
+     */
+    const generateMenuItems = wallets =>
+      wallets.map(wallet => {
+        return {
+          id: generateMenuItemId(wallet.id),
+          title: <WalletName wallet={wallet} />,
+          onClick: () => {
+            setActiveWallet(wallet.id)
+            history.push(`/home/wallet/${wallet.id}`)
+          },
+        }
+      })
+
+    return (
+      <Box p={2} {...rest}>
+        <WalletGroupHeader>{title}</WalletGroupHeader>
+        <Menu items={generateMenuItems(wallets)} selectedItem={generateMenuItemId(activeWallet)} />
```

Comment:
Not sure if it's a problem, but `activeWallet` can be undefined, so `generateMenuItemId` will also generate something weird